components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/device/DeviceEndpoint.java

```diff
@@ -0,0 +1,137 @@
+        String redirectionUriComplete = redirectionUri + "?user_code=" + userCode;
+        Long expiresIn = 3600000L;
+        int interval = 5000;
+        OAuthResponse errorResponse;
+
+        if (clientId != null) {
+            if (validateClientId(clientId)) {
+
+                DeviceFlowPersistenceFactory.getInstance().getDeviceFlowDAO().insertDeviceFlow(deviceCode, userCode,
+                        clientId, scope, expiresIn);
+
+                OAuthResponse deviceResponse =
+                        OAuthResponse.status(HttpServletResponse.SC_ACCEPTED).setParam(Constants.DEVICE_CODE,
+                                deviceCode).setParam(Constants.USER_CODE, userCode).setParam(Constants.VERIFICATION_URI,
+                                redirectionUri).setParam(Constants.VERIFICATION_URI_COMPLETE, redirectionUriComplete).
+                                setParam(Constants.EXPIRES_IN, String.valueOf(expiresIn / 1000))
+                                .setParam(Constants.INTERVAL, String.valueOf(interval / 1000)).buildJSONMessage();
+                Response.ResponseBuilder respBuilder = Response.status(response.getStatus());
+                return respBuilder.entity(deviceResponse.getBody()).build();
+
+            } else {
+                response.setStatus(HttpServletResponse.SC_UNAUTHORIZED);
+
+                try {
+                    errorResponse = OAuthASResponse
+                            .errorResponse(response.getStatus())
+                            .setError(DeviceErrorCodes.UNAUTHORIZED_CLIENT)
+                            .setErrorDescription("No registered client with the client id.")
+                            .buildJSONMessage();
+
+                    Response.ResponseBuilder respBuilder = Response.status(response.getStatus());
+                    return respBuilder.entity(errorResponse.getBody()).build();
+                } catch (OAuthSystemException e) {
+
+                    if (log.isDebugEnabled()) {
+                        log.debug("Error building errorResponse due to:", e);
+                    }
+                }
+            }
+        } else {
+            response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+            try {
+                errorResponse = OAuthASResponse
+                        .errorResponse(response.getStatus())
+                        .setError(DeviceErrorCodes.INVALID_REQUEST)
+                        .setErrorDescription("Request missing required parameters")
+                        .buildJSONMessage();
+
+                Response.ResponseBuilder respBuilder = Response.status(response.getStatus());
+                return respBuilder.entity(errorResponse.getBody()).build();
+            } catch (OAuthSystemException e) {
+
+                if (log.isDebugEnabled()) {
+                    log.debug("Error building errorResponse due to:", e);
+                }
+            }
+        }
+        return null;
+    }
+
+    /**
+     * validate the client id
```

Comment:
When explaining the purpose of a function, Start the sentence with a capital letter and end with a full stop. Fix this in other applicable places.